api/tests/unit/domain/models/UserLogin_test.js

```diff
@@ -285,17 +285,41 @@ describe('Unit | Domain | Models | UserLogin', function () {
   });
 
   describe('#anonymize', function () {
-    it('anonymizes user login info', function () {
-      // given
-      const userLogin = new UserLogin({ id: 1, temporaryBlockedUntil: new Date(), blockedAt: new Date() });
+    context('when there is no lastLoggedAt', function () {
+      it('anonymizes user login info', function () {
+        // given
+        const userLogin = new UserLogin({ id: 1, temporaryBlockedUntil: new Date(), blockedAt: new Date() });
 
-      // when
-      const result = userLogin.anonymize();
+        // when
+        const result = userLogin.anonymize();
 
-      // then
-      expect(result.id).to.be.equal(1);
-      expect(result.temporaryBlockedUntil).to.be.null;
-      expect(result.blockedAt).to.be.null;
+        // then
+        expect(result.id).to.be.equal(1);
+        expect(result.temporaryBlockedUntil).to.be.null;
+        expect(result.blockedAt).to.be.null;
+        expect(result.lastLoggedAt).to.be.null;
+      });
+    });
+
+    context('when there is a lastLoggedAt', function () {
+      it('anonymizes user login info', function () {
+        // given
+        const userLogin = new UserLogin({
+          id: 1,
+          temporaryBlockedUntil: new Date(),
+          blockedAt: new Date(),
+          lastLoggedAt: new Date(),
+        });
+
+        // when
+        const result = userLogin.anonymize();
+
+        // then
+        expect(result.id).to.be.equal(1);
+        expect(result.temporaryBlockedUntil).to.be.null;
+        expect(result.blockedAt).to.be.null;
+        expect(result.lastLoggedAt).to.be.an.instanceOf(Date);
+      });
```

Comment:
La distinction des 2 contextes apporte peu ici.

Il faudrait modifier le test pour s'assurer que la généralisation des dates a bien été appliquée sur les 3 dates (`createdAt`, `updatedAt`, `lastLoggedAt`)

Reply:
Ah non, j'ai compris le test à la seconde relecture ^^
Je pense qu'il faut le renommer et vérifier uniquement ce qui nécessaire dans le test.

```js
context('when there is a lastLoggedAt', function () {
  it('keeps the lastLoggedAt and generalizes it', function () {
    // given
    const userLogin = new UserLogin({
      id: 1,
      lastLoggedAt: new Date(),
    });

    // when
    const result = userLogin.anonymize();

    // then
    expect(result.lastLoggedAt).to.be.an.instanceOf(Date);
    expect(result.lastLoggedAt).to.be.equal('THEDATEGENERALIZED');
  });
});
```

Reply:
Tout à fait d'accord pour renommer et vérifier uniquement ce qui nécessaire dans ce test :+1: 

Mais l'assertion `expect(result.lastLoggedAt).to.be.equal('THEDATEGENERALIZED');` testerait l'implémentation de la généralisation de la date, qui est testée par ailleurs. Aussi je propose de remplacer cette suggestion d'assertion par une autre assertion qui vérifie que la `result.lastLoggedAt` a été modifiée, mais sans vérifier sa valeur : 

```javascript
expect(result.lastLoggedAt).to.not.equal(someDate);
```

(je viens de pousser cette modif avec des nouveaux commits à l'instant)

Reply:
Je ne suis pas vraiment d'accord avec: "testerait l'implémentation de la généralisation de la date". On ne teste pas l'implémentation mais le résultat (ce qui est attendu d'un point de vue métier pour l'anonymisation. On peut changer l'implémentation de la généralisation, elle n'affecte pas le résultat (si les règles métier n'ont pas changées).

Reply:
Par ailleurs, il faut que ce test échoue si quelqu'un change l'utilitaire de généralisation, car la règle métier est portée par cette fonction `anonymize`